src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt

```diff
@@ -0,0 +1,215 @@
+        displayName,
+        vals.map { it.first }.toTypedArray(),
+        vals.indexOfFirst { it.second == defaultValue }.takeIf { it != -1 } ?: 0,
+    ) {
+        fun selectedValue() = vals[state].second
+    }
+
+    private class OrderByFilter(defaultOrder: String? = null) : SelectFilter(
+        "Order by",
+        arrayOf(
+            Pair("Recently", "updated_at"),
+            Pair("New", "created_at"),
+            Pair("Hot", "views"),
+        ),
+        defaultOrder,
+    )
+
+    override fun getFilterList(): FilterList = FilterList(
+        Filter.Header("Ignored when using text search"),
+        Filter.Separator(),
+        OrderByFilter(),
+    )
+
+    // =========================== Manga Details ============================
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        Observable.just(manga)
+
+    override fun mangaDetailsParse(response: Response): SManga =
+        throw UnsupportedOperationException()
+
+    // ============================== Chapters ==============================
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val id = manga.url.substringAfterLast("-")
+        return GET("$apiUrl/$id/childs", apiHeaders)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        return response.parseAs<ChapterListDto>().data.map { it.toSChapter() }
+    }
+
+    // =============================== Pages ================================
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+        val scriptData = document.selectFirst("script:containsData(initialChapter)")?.data()
+            ?: throw Exception("Unable to find image data")
+        val data = json.decodeFromString<PagesDto>(scriptData).props.pageProps.initialChapter.images
+        return data.map { img ->
+            Page(img.index, imageUrl = img.url)
+        }
+    }
+
+    override fun imageUrlParse(response: Response) = throw UnsupportedOperationException()
+
+    override fun imageRequest(page: Page): Request = super.imageRequest(page).newBuilder()
+        .header("Accept", "image/avif,image/webp,*/*")
+        .build()
+
+    // ============================= Utilities ==============================
+
+    private fun ChapterListDto.ChapterDto.toSChapter(): SChapter = SChapter.create().apply {
+        name = this@toSChapter.name
+        chapter_number = index.toFloat()
+        url = "/read/$slug-$chapterId"
+        createdAt?.let { date_upload = parseDate(it) }
+    }
+
+    private fun ListDto.EntryDto.toSManga(): SManga = SManga.create().apply {
+        title = name
+        url = "/manga/$slug-$mangaId"
+        thumbnail_url = image
+        description = buildString {
+            this@toSManga.description?.let {
+                append(it)
+                altName?.let { append("\n\n") }
+            }
+            altName?.let { append("Alternative name(s): $it") }
```

Comment:
```suggestion
            }
            altName?.let {
                append("\n\n")
                append("Alternative name(s): ")
                append(it)
            }
```
Using string interpolation will create a StringBuilder internally.